src/c-writer.cc

```diff
@@ -922,33 +922,22 @@ void CWriter::WriteFuncTypes() {
 }
 
 void CWriter::WriteTags() {
-  if (module_->tags.empty()) {
-    Write("static void init_tags(void) ", OpenBrace(), CloseBrace(), Newline());
-    return;
+  for (auto it = module_->tags.cbegin() + module_->num_tag_imports;
+       it != module_->tags.cend(); ++it) {
```

Comment:
The pattern used in other functions of this type (e.g. WriteFuncDeclarations) is to use:

```
for (const Func* func : module_->funcs) {                                         
    bool is_import = func_index < module_->num_func_imports;                        
    if (!is_import) {                                                               
       .. do something                                            
    }                                                                               
    ++func_index;                                                                   
}
```   

Even if `func_index` isn't actually used.   Maybe use this same pattern and then we can consider a separate patch to change to this iterator pattern instead everywhere if you think its better.

Reply:
Done.